File: lint-arwaky/src-python/contract/http_provider_port.py
"""http_provider_port — Port interface for HTTP operations."""

from abc import ABC, abstractmethod
from ..taxonomy import ResponseData, TransportUrlVO, Timeout, TransportError


class IHttpProviderPort(ABC):
    """Abstraction for HTTP operations."""

    @abstractmethod
    def get(
        self, url: TransportUrlVO, timeout: Timeout = Timeout(value=2000)
    ) -> ResponseData | TransportError:
        """Performs a GET request."""
        ...

File: lint-arwaky/src-rust/contract/http_provider_port.rs
use crate::taxonomy::{ContentString, ResponseData, Timeout, TransportUrlVO};

#[async_trait::async_trait]
pub trait IHttpProviderPort: Send + Sync {
    async fn get(
        &self,
        url: TransportUrlVO,
        timeout: Option<Timeout>,
    ) -> Result<ResponseData, String>;

    async fn post(
        &self,
        url: TransportUrlVO,
        body: ContentString,
        timeout: Option<Timeout>,
    ) -> Result<ResponseData, String>;
}
